core/src/main/scala/dagr/core/reporting/ExecutionLogger.scala

```diff
@@ -0,0 +1,210 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.core.reporting
+
+import com.fulcrumgenomics.commons.CommonsDef.{FilePath, SafelyClosable}
+import com.fulcrumgenomics.commons.io.Io
+import dagr.core.reporting.ExecutionLogger.{Definition, Relationship, Status}
+import dagr.core.reporting.ReportingDef._
+import dagr.core.tasksystem.Task
+import dagr.core.tasksystem.Task.{TaskStatus, TaskInfo => RootTaskInfo}
+
+import scala.collection.mutable
+
+object ExecutionLogger {
+
+  val Separator: String = ","
+
+  trait ToStringIsSimpleNameUpperCase extends Product {
+    override def toString: String = {
+      val name = this.getClass.getSimpleName.replaceFirst("[$].*$", "").toUpperCase
+      (Seq(name) ++ this.productIterator.map(_.toString)).mkString(Separator)
+    }
+  }
+
+  object Definition {
+    val Name: String        = "DEFINITION"
+    val Header: Seq[String] = Seq("SIMPLE_NAME", "NAME", "CODE", "PARENT_CODE", "CHILD_NUMBER")
+
+    /** Removes various characters from the simple class name, for scala class names. */
+    def getSimpleName(task: Task): String = {
+      task.getClass.getSimpleName.replaceFirst("[$].*$", "")
+    }
+
+    def buildRootDefinition(root: Task): Definition = {
+      Definition(root, -1, 0)
+    }
+
+    def isRootDefinition(definition: Definition): Boolean = {
+      definition.parentCode == -1 && definition.childNumber == 0
+    }
+
+    def apply(line: String): Definition = {
+      line.split(Separator).toList match {
+        case Name :: simpleName :: name  :: code :: parentCode :: childNumber :: Nil =>
+          Definition(simpleName, name, code.toInt, parentCode.toInt, childNumber.toInt)
+        case _ =>
+          throw new IllegalArgumentException(s"Could not parse line: $line")
+      }
+    }
+    def apply(task: Task, parentCode: Int, childNumber: Int): Definition = {
+      Definition(getSimpleName(task), task.name, task.hashCode, parentCode, childNumber)
+    }
+  }
+
+  /** Attempts to uniquely identify a task.
+    * @param simpleName the simple class name for the task
+    * @param name the actual name of the task (not considered, for debugging only)
+    * @param code a unique number for this definition
+    * @param parentCode a unique number fo the parent's definition
+    * @param childNumber the 0-based index of the child relative to all children of the parent
+    */
+  case class Definition(simpleName: String, name: String, code: Int, parentCode: Int, childNumber: Int) extends ToStringIsSimpleNameUpperCase {
+    /** Returns true if the the other definition represents the same task as the this definition.  Assumes that
+      * the parents have the same number of child tasks. */
+    def equivalent(that: Definition): Boolean = {
```

Comment:
@tfenne this is the where we determine two tasks are "equivalent" across executions.  You'll likely find that it is too conservative, so any ideas would be helpful.